File: OrderServer/OrderServer.API/OrderServer.Shared/Cache/ICacheService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OrderServer.Shared.Cache
{
    public interface ICacheService
    {
        Task<TCache?> GetAsync<TCache>(string cacheKey);
        Task CreateAsync<TCache>(string cacheKey, TCache data, DateTimeOffset ttl);
        Task<bool> DeleteAsync(string cacheKey);
    }
}

File: PaymentServer/PaymentServer.API/packages/common/src/caching/ICacheService.ts
export interface ICacheService {
    keys(key: string): Promise<string[]>;
    set<TCache>(key: string, data: TCache, ttl: number): Promise<void>;
    get<TCache>(key: string, ctor: new (...args: any) => TCache): Promise<TCache | null>;
    remove(key: string): Promise<boolean>;
}